### PR TITLE
Update code to support only Record and String array of array type in CSV read

### PR DESCRIPTION
**Pull request by sahanHe**
## Purpose
Fixes: [3669](https://github.com/ballerina-platform/ballerina-standard-library/issues/3669)
## Checklist
- [x] Linked to an issue
- [x] Updated the changelog
- [x] Added tests
- [ ] ~Updated the spec~
- [x] Checked native-image compatibility

### CODE REVIEW

File: native/src/main/java/io/ballerina/stdlib/io/nativeimpl/RecordChannelUtils.java [hunk 1]
```diff
@@ -186,6 +186,8 @@ public static Object getAllRecords(BObject channel, int skipHeaders, BTypedesc t
                 }
                 Object[] out = outList.toArray();
                 return ValueCreator.createArrayValue(out, TypeCreator.createArrayType(describingType));
+            } else {
+                return IOUtils.createError("Unsupported mapping type detected. ");
```
Comment (niveathika) [inline native/src/main/java/io/ballerina/stdlib/io/nativeimpl/RecordChannelUtils.java:190]: Only ' ' supported, but found ''
Lets format this as above

Comment (sahanHe) [reply native/src/main/java/io/ballerina/stdlib/io/nativeimpl/RecordChannelUtils.java]: updated in [e6b9e1a](https://github.com/ballerina-platform/module-ballerina-io/pull/898/commits/e6b9e1aed7f24cc77568f0fb702efd18da79d752)

**Review by daneshk — APPROVED** [review]
LGTM

File: native/src/main/java/io/ballerina/stdlib/io/nativeimpl/RecordChannelUtils.java [hunk 2]
```diff
@@ -174,7 +174,7 @@ public static Object getAllRecords(BObject channel, int skipHeaders, BTypedesc t
                 }
                 Object[] out = outList.toArray();
                 return ValueCreator.createArrayValue(out, TypeCreator.createArrayType(describingType));
-            } else {
+            } else if (describingType.getTag() == TypeTags.ARRAY_TAG) {
```
Comment (niveathika) [inline native/src/main/java/io/ballerina/stdlib/io/nativeimpl/RecordChannelUtils.java:177]: Here only the array type is checked not string[]. How does the function behave if I give a return type of boolean[]. If this is addressed it is fine.

File: native/src/main/java/io/ballerina/stdlib/io/nativeimpl/RecordChannelUtils.java [hunk 3]
```diff
@@ -186,6 +186,9 @@ public static Object getAllRecords(BObject channel, int skipHeaders, BTypedesc t
                 }
                 Object[] out = outList.toArray();
                 return ValueCreator.createArrayValue(out, TypeCreator.createArrayType(describingType));
+            } else {
+                return IOUtils.createError(String.format("Only 'string[]' and record{} are supported, " +
```
Comment (niveathika) [inline native/src/main/java/io/ballerina/stdlib/io/nativeimpl/RecordChannelUtils.java:190]:
```suggestion
                return IOUtils.createError(String.format("Only 'string[]' and 'record {}' types are supported, " +
```

File: native/src/main/java/io/ballerina/stdlib/io/nativeimpl/RecordChannelUtils.java (hunk 2 above)
Comment (sahanHe) [reply native/src/main/java/io/ballerina/stdlib/io/nativeimpl/RecordChannelUtils.java:177]: if we give the return type as boolean[] it returns the following error message `incompatible type for parameter 'returnType' with inferred typedesc value: expected 'typedesc<(string[]|map<anydata>)>', found 'typedesc<boolean>'`. this is handles in [here](https://github.com/sahanHe/module-ballerina-io/blob/e6b9e1aed7f24cc77568f0fb702efd18da79d752/ballerina/file_csv_io.bal#L28) as the expected type is `string[]|map<anydata>[]`

File: native/src/main/java/io/ballerina/stdlib/io/nativeimpl/RecordChannelUtils.java (hunk 3 above)
Comment (sahanHe) [reply native/src/main/java/io/ballerina/stdlib/io/nativeimpl/RecordChannelUtils.java]: updated in [ec66f19](https://github.com/ballerina-platform/module-ballerina-io/pull/898/commits/ec66f19f98d5e5c2a8590e2da49696154ce9a8f4)

File: native/src/main/java/io/ballerina/stdlib/io/nativeimpl/RecordChannelUtils.java (hunk 2 above)
Comment (niveathika) [reply native/src/main/java/io/ballerina/stdlib/io/nativeimpl/RecordChannelUtils.java:177]: Ack

**Review by niveathika — APPROVED** [review]
LGTM

**Pull request merged**